chirp/drivers/kenwood_live.py

```diff
@@ -168,6 +169,53 @@ def iserr(result):
     return result in ["N", "?"]
 
 
+class KenwoodSettingProgrammableVFO(RadioSettingValueString):
+
+    """A string setting"""
+
+    def __init__(self, minval, maxval, current):
+        minlen = len("%d" % minval)
+        maxlen = len("%d" % maxval)
+        start, end = current.split(',', 1)
+        current = "%d-%d" % (int(start), int(end))
+        LOG.debug("Current = '%s'" % current)
+        RadioSettingValueString.__init__(self, minlen * 2 + 1,
+                                         maxlen * 2 + 1, current,
+                                         False, "0123456789-")
+        self._minval = minval
+        self._maxval = maxval
+        self.set_value(current)
+
+    def validate(self):
+        value = self.get_value()
+        LOG.debug("Validating... %s" % value)
+        try:
+            start, end = value.split("-", 1)
+            start = int(start)
+            end = int(end)
+            if start > end: # TODO: Set current value back on error...
+                raise errors.RadioError("Start must be less than or equal "
+                                        "to end")
+            if (start < self._minval):
+                raise errors.RadioError("Start must be greater than or equal "
+                                        "to %d" % self._minval)
+            if (start > self._maxval):
+                raise errors.RadioError("Start must be less than or equal "
+                                        "to %d" % self._maxval)
+            if (end < self._minval):
+                raise errors.RadioError("End must be greater than or equal "
+                                        "to %d" % self._minval)
+            if (end > self._maxval):
+                raise errors.RadioError("End must be less than or equal to " +
+                                        "%d" % self._maxval)
+        except errors.RadioError:
+            raise
+        except:
+            raise errors.RadioError("Programmable VFO must be Start-End " +
+                                    "format (ie: 144-147)")
+        return "%05d,%05d" % (start, end)
```

Comment:
Er, this format is different than what you say it needs to be above in the error and in your splitting. I feel like it would be better to expose this as two settings, an upper and lower, like the other radios do...

Reply:
When modifying on the radio, it's presented as "XXX-XXXMHz", and you can change between the two fields.  Since the two values are set together with the same command, I wanted it to stay together since it seems that every time you change a setting, all the settings that have been changed are also updated, so I didn't want weird ordering "stuff" to happen.

Reply:
Okay, so you'll get *all* the settings at once, always. So I think you can handle it in a stable way even if they're separate. But yeah, let me think on this after I play with it on a D700.